platform/ios/src/MGLMapView.mm

```diff
@@ -1299,8 +1304,122 @@ - (void)didMoveToSuperview
     [super didMoveToSuperview];
 }
 
+
+- (UIInterfaceOrientationMask)applicationSupportedInterfaceOrientations {
+    static UIInterfaceOrientationMask orientationMask = UIInterfaceOrientationMaskAll;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        
+        // "The system intersects the view controller's supported orientations with
+        // the app's supported orientations (as determined by the Info.plist file or
+        // the app delegate's application:supportedInterfaceOrientationsForWindow:
+        // method) and the device's supported orientations to determine whether to rotate.
+        
+        UIApplication *application = [UIApplication sharedApplication];
+        
+        if (self.window && [application.delegate respondsToSelector:@selector(application:supportedInterfaceOrientationsForWindow:)]) {
+            orientationMask = [application.delegate application:application supportedInterfaceOrientationsForWindow:self.window];
+            return;
+        }
+        
+        // No application delegate
+        
+        NSString *key;
+        
+        switch ([[UIDevice currentDevice] userInterfaceIdiom]) {
+            case UIUserInterfaceIdiomPhone:
+                key = @"UISupportedInterfaceOrientations~iphone";
```

Comment:
Is it always the case that this key has `~iphone` (or `~ipad`) appended? From reading [the key docs](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/iPhoneOSKeys.html#//apple_ref/doc/uid/TP40009252-SW10), it seems like a non-universal app may use a plain `UISupportedInterfaceOrientations`.

Reply:
Ah, I see — the unappended case is checked later if the appended case fails. I wonder if there’s a good way to consolidate these.

Reply:
Yep, this code isn't needed at all! I forgot the system takes care of this for you: 

> When the system searches for a key in your app’s Info.plist file, it chooses the key that is most specific to the current device and platform. If it does not find a qualified key, it looks for one without qualifiers.